File: src/Array/python/PascalTriangle.py
class Solution(object):
    def generate(self, numRows):
        """
        :type numRows: int
        :rtype: res[res[int]]
        """
        res = [0]*numRows
        if numRows == 0:
            res =[]
            return res
        if numRows == 1:
            res = [[1]]
            return res
        else :
            res[0] = [1]
            res[1] = [1,1]
            for i in range(2,numRows):
                res[i] = [0]*(i+1)
                res[i][0] = res[i][i] = 1
                for j in range(1,i):
                    res[i][j] = res[i-1][j-1]+res[i-1][j]
            print res
            return res

Test = Solution()
Test.generate(5)

File: src/qiuyi/Array/PascalTriangle.cpp
class Solution {
public:
	vector<vector<int>> generate(int numRows) {
		vector<vector<int>> triangle;

		for (int i = 0; i < numRows; i++) {
			vector<int> temp;
			for (int j = 0; j <= i; j++) {
				if (j == 0 || j == i)
					temp.push_back(1);
				else
					temp.push_back(triangle[i-1][j-1]+triangle[i-1][j]);
			}
			triangle.push_back(temp);
		}
		return triangle;
	}
};